sdk/src/main/java/dev/cdevents/CDEvents.java

```diff
@@ -116,35 +109,114 @@ public static CDEvent cdEventFromJson(String cdEventJson) {
+        }
+    }
 
-            if (!errors.isEmpty()) {
-                log.error("CDEvent Json validation failed against schema URL {}", cdEvent.schemaURL());
-                log.error("CDEvent Json validation failed with errors {}", errors);
-                return false;
+    /**
+     * Creates customCDEvent from Json string and validates against context and official schemas.
+     * @param customCDEventJson Json string of customCDEvent class type <T>
+     * @param <T> customCDEvent class
+     * @param eventClass custom CDEvent class of type <T>
+     * @param validateContextSchema true If validation needed against context.schemaUri
+     * @return CDEvent
+     */
+    public static <T extends CDEvent> T customCDEventFromJson(String customCDEventJson, Class<T> eventClass, boolean validateContextSchema) {
+        try {
+            T cdEvent = new ObjectMapper().readValue(customCDEventJson, eventClass);
+            if (!validateCustomCDEvent(cdEvent, validateContextSchema)) {
+                throw new CDEventsException("Custom CDEvent validation failed.");
             }
+            return cdEvent;
         } catch (JsonProcessingException e) {
-            throw new CDEventsException("Exception occurred while validating CDEvent json with schema file ", e);
+            throw new CDEventsException("Exception occurred while processing cdEventJson with the event class " + eventClass.getName(), e);
+        }
+    }
+
+    /**
+     * Validates the custom CDEvent against the official and context schemas.
+     * @param customCDEvent custom CDEvent to validate
+     * @param validateContextSchema true to validate custom CDEvent against context schema
+     * @return valid cdEvent
+     */
+    public static boolean validateCustomCDEvent(CDEvent customCDEvent, boolean validateContextSchema) {
+        if (validateContextSchema) {
+            return validateWithContextSchemaUri(customCDEvent) && validateWithOfficialCustomSchema(customCDEvent);
+        } else {
+            return validateWithOfficialCustomSchema(customCDEvent);
+        }
+    }
+
+    /**
+     * Validates the custom CDEvent against the provided context Schema URI.
+     * @param customCDEvent custom CDEvent to validate
+     * @return valid cdEvent
+     */
+    public static boolean validateWithContextSchemaUri(CDEvent customCDEvent) {
+        if (customCDEvent.customSchemaUri() == null) {
+            log.error("Context schemaUri does not exist, required for custom schema validation.");
+            throw new CDEventsException("Context schemaUri does not exist.");
+        }
+        log.info("Validate custom CDEvent against context.schemaUri - {}", customCDEvent.customSchemaUri());
+        JsonSchemaFactory factory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012);
+        JsonSchema jsonSchema = factory.getSchema(customCDEvent.customSchemaUri());
```

Comment:
I think we should not promote a model where for every message the SDK reaches out to some internet location to fetch the schema. The way I was thinking of implementing this for the sdk-go is to let user register a custom schema, so that it's loaded once and re-used from memory, and then give users an option to decide what to do if the custom `schemaURI` is not in the in-memory schema cache, i.e. fail or try to download the schema, with the former as default.

Reply:
yes agree, will do the required changes in a separate PR. Removed this implementation from current PR and created issue to work on https://github.com/cdevents/sdk-java/issues/84